CONTRIBUTING.txt

```diff
@@ -0,0 +1,98 @@
+    5. Code submissions MUST adhere to the code style guidelines of the project if these are defined.
+    6. If a PR contains multiple commits, each commit MUST compile cleanly and add value to the history of the project. Checkpoint commits should be squashed.
+    7. A PR SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
+    8. A PR MUST compile cleanly and pass project self-tests on all target platforms.
+    9. Each commit message in a PR MUST succinctly explain what the commit achieves.
+    10. The PR Description MUST consist of a single short (less than 50 characters) line stating the problem (“Problem: …") being solved, followed by a blank line and then the proposed solution (“Solution: …").
+    11. A “Valid PR” is one which satisfies the above requirements.
+
+6. Process
+
+    1. Change on the project follows the pattern of accurately identifying problems and applying minimal, accurate solutions to these problems.
+    2. To request changes, a user (end user, contributor, maintainer, or administrator) logs an issue on the project Platform issue tracker.
+    3. The user or contributor should write the issue by describing the problem they face or observe. Links to the forum or other resources are permitted but the issue should be complete and accurate and should not require the reader to visit the forum or any other platform to understand what is being described.
+    4. Issues should strive to describe the minimum acceptable condition.
+    5. Issues should focus on user tasks and avoid comparisons to other software solutions.
+    6. The user or Contributor SHOULD seek consensus on the accuracy of their observation and the value of solving the problem.
+    7. To submit a solution to a problem, a Contributor SHALL create a pull request back to the project.
+    8. Contributors and Maintainers SHALL NOT commit changes directly to the target branch.
+    9. To discuss a proposed solution, users MAY comment on the platform pull request. Forum conversations regarding the solution should be discouraged and conversation redirected to the Pull Request or the related issue.
+    10. To accept or reject a Pull Request, a Maintainer SHALL use the platform interface.
+    11. Maintainers SHOULD NOT merge their own PRs except:
+      - in exceptional cases, such as non-responsiveness from other Maintainers for an extended period (more than 1-2 days).
+      - If the maintainer is also the primary developer of the workbench or subsystem.
+
+    12. Maintainers SHALL merge valid PRs from other Contributors rapidly.
+    13. Maintainers MAY, at their discretion merge PRs that have not met all criteria to be considered valid to:
+        - (a) ending fruitless discussions
+        - (b) capturing toxic patches in the historical record
+        - (c) engaging with the Contributor on improving their patch quality.
+    14. Maintainers SHALL NOT make value judgments on correct patches.
```

Comment:
> What is a "value judgement"?

A judgment about the value of the contribution.  FOSS projects can suffer from holy wars where someone wants to make a change but someone else doesn't think that change is worthwhile. (GUI stuff can often go this way)

The intent of this 'rule' is to avoid those altogether.  Otherwise, the person with the most authority wins even if their judgment is flawed.

This really only works if there's a good 1:1 relationship between problems and solutions.  Then the 'value' is discussed thoroughly in the 'problem', ideally narrowing and focusing it.  Then the solution is only judged on whether it resolves the problem.

Reply:
I see. Thanks.